.github/workflows/test.yml

```diff
@@ -26,19 +26,51 @@ jobs:
               exit 1
           fi
 
-      - name: make build
+      - name: make build-arm64
         run: |
-          sudo -u testbot bash -lc 'make build'
+          sudo -u testbot bash -lc 'make build-arm64'
 
+  test_32_bit: 
+      runs-on: buildjet-4vcpu-ubuntu-2204-arm
+      container: ghcr.io/viam-modules/raspberry-pi:arm32
+      timeout-minutes: 30
+  
+      steps:
+        - uses: actions/checkout@v3
+  
+        - name: Verify no uncommitted changes from make lint
+          run: |
+            git init
+            git add .
+            chown -R testbot:testbot .
+            sudo -u testbot bash -lc 'make lint'
+            if [ -n "$GEN_DIFF" ]; then
+                echo '"make lint" resulted in changes not in git' 1>&2
+                git status
+                exit 1
+  
+        - name: Build for ARM32
+          run: |
+            sudo -u testbot bash -lc 'make build-arm32'
+  
   test_on_pi:
     # The following steps are run on an external runner "rpibull"
     runs-on: pi-4
     timeout-minutes: 30
 
+    strategy:
+      matrix:
+        architecture: [arm64, armhf]  # Run tests for both 32-bit and 64-bit binaries
+  
     steps:
       - uses: actions/checkout@v3
-      - name: build
-        run: make build
+      - name: Build for ${{ matrix.architecture }}
+        run: |
+          if [ "${{ matrix.architecture }}" = "armhf" ]; then
+            make build-arm32
+          else
+            make build-arm64
+          fi
 
-      - name: run tests
+      - name: Run tests for ${{ matrix.architecture }}
```

Comment:
I don't think this is doing what you think it is. The "matrix" just sets a matrix of variables. You still have to make use of them. Just changing the name for the test step doesn't mean the test step is actually going to run on that architecture. So the test is going to run on the same runner both times, with the same (likely 64bit) architecture.